EIPS/eip-2212.md

```diff
@@ -0,0 +1,271 @@
+
+## Abstract
+
+The advent of decentralised lending brought to the web3 ecosystem a new primitive for designing business models. We herein describe a smart contract interface that accepts deposits ("stakes") in [ERC-20](https://eips.ethereum.org/EIPS/eip-20) tokens. These deposits are immediately lent on the market to earn interest either for just the owner of the contract, or for both them and the user. We assume that users get access to a product or a service for as long as they keep staking tokens.
+
+## Motivation
+
+While most web3 business models rely on paying a percentage fee on transfers, or committing to a monthly subscription, we aim to flip the model on its head. Users are their own banks now, so, instead of charging them, we make them stake tokens that can be claimed back, in full, at any point in time. We view this as a win-win-win scenario for users, dapp creators and lending protocols.
+
+It is worth it to mention PoolTogether, a "no-loss lottery" that launched recently. It works by having participants deposit money in their contracts, locking it up for a while and earning interest through Compound. Ultimately, they choose a lucky winner and return the deposits back to all other players. PoolTogether takes a 10% commission on the prize.
+
+We wrote this specification because interest earning stakes are a powerful financial instrument and a community-vetted standard and implementation will reduce the cost of replication.
+
+### Use Cases
+
+- Software products and services
+- Charity pools
+- Games
+- Gym memberships
+
+## Specification
+
+Assigning an owner is required for this spec to make sense, thus we assume that a proper implementation would use a
+well-known contract such as OpenZeppelin's
+[Ownable](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/ownership/Ownable.sol). For
+brevity, we haven't included any method for assigning or transferring ownership.
+
+Every time we refer to cTokens, we mean Compound Tokens, as defined in the official
+[documentation](https://compound.finance/ctokens). These are ERC-20 tokens with an interest accrual feature built in.
+
+### Methods
+
+#### owner
+
+Returns the owner of the contract.
+
+```solidity
+function owner() external view returns (address);
+```
+
+#### fee
+
+Returns the fee as a percentage value scaled by 1e18.
+
+```solidity
+function fee() external view returns (uint256);
+```
+
+#### balanceOf
+
+Returns the stake that consists of the initial deposit plus the interest accrued over time.
+
+SHOULD revert if `staker` doesn't have an active stake. SHOULD revert if `tokenAddress` doesn't match either the cToken or the underlying of the stake.
+
+```solidity
+function balanceOf(address staker, address tokenAddress) external view returns (uint256);
```

Comment:
Since the ABI encoding ignores the parameter names, this may be something prone to collisions.